Add MP Promo to Home page

mmmavis opened the pull request:
This fixes #129

mmmavis committed:
Fixed #129 - created <Promo> component and added MP Promo to Home page

mmmavis commented on src/components/promo/promo.less:
```diff
@@ -0,0 +1,37 @@
+    height: auto;
+    margin: 1rem 0;
+    padding: 20px 50px;
+  }
```
I found some [`button` rulesets in `splash.less`](https://github.com/mozilla/webmaker-desktop/blob/develop/src/pages/splash/splash.less#L27-L48) but I wasn't sure what those were for (there wasn't any `<button`> in `splash.jsx`)

Other than the rules provided in [guideline](https://cloud.githubusercontent.com/assets/1119821/8234436/ef93135a-15a9-11e5-9ad6-b1fa6c2beaad.png) I added 

``` css
width: auto;
height: auto;
margin: 1rem 0;
padding: 20px 50px;
```

just to override what's already in `splash.less`

mmmavis commented:
@xmatthewx this is ready for review! :dancers: (I just don't know who to assign this to though :eyes: )
Just a note that after looking over the existing code I'm still not sure what unit I should stick to... so there is a mixture of `px` and `rem`.  Happy to change them if necessary :smile_cat:

mmmavis committed:
removed an extra comma

gvn commented on src/components/promo/promo.jsx:
```diff
@@ -0,0 +1,29 @@
+               width="100" />
+          <h3>{this.props.head}</h3>
+          <h4>{this.props.subhead}</h4>
+          <button onClick={this.handleBtnClick}>{this.props.ctaText}</button>
```
This could just be a standard anchor element instead of forcing a button to behave like one.

mmmavis replied:
Yea... I got really confused about the [`.btn`](https://github.com/mozilla/webmaker-desktop/blob/develop/src/components/button/button.less) and [`button stylings`](https://github.com/mozilla/webmaker-desktop/blob/develop/src/pages/splash/splash.less#L27-L48)(<--- there's no `<button>` in `splash.jsx` either :no_mouth: ) in this repo.  Really don't wanna override all the existing stylings... but maybe that's the way it should be?

xmatthewx replied:
We were initially collecting emails before shifting to the play store CTA. Button styling might be legacy from that.

gvn replied:
We generally use a .btn class that can be applied to any element type to make it look like a button and then .btn-VARIANT to override and extend that class.

gvn commented on src/pages/splash/splash.less:
```diff
@@ -248,6 +248,18 @@
+   */
+  #mp-promo {
+    .promo {
+      .promoMixin(@vikingYellow);
```
Why is this a mixin if you're only using it once?

mmmavis replied:
From the [guideline](https://github.com/mozilla/webmaker-desktop/issues/79#issuecomment-113186712) it feels like this will be applied to any future promo banners, not just the Maker Party one we have in this PR? :eyes:

gvn replied:
But if all you're changing is the color, you can just change that value when the design changes, right? Or are there going to be multiple banners up at a time with different colors?

mmmavis replied:
My assumption is there could be more than one, that's why a reusable `<Promo>` component could come in handy, but I could be wrong!

(not sure who to flag...)@xmatthewx @vazquez ? :hamster:

xmatthewx replied:
Yes. Promos will change periodically. They will use the same core styles and select from a small set of background color options. Some day, they might appear on pages other than the splash page.

gvn replied:
Ok, in that case I guess it's fine to leave it a mixin...

gvn commented on src/components/promo/promo.less:
```diff
@@ -0,0 +1,37 @@
+    font-size: 2.25rem;
+    margin-bottom: 1rem;
+    font-weight: 400;
+    margin: 20px 0 10px 0;
```
You're overriding the previous margin-bottom, so you can just get rid of it.

mmmavis replied:
oh yea oops!

gvn commented:
A few tweaks, but looking really good!

mmmavis committed:
styling tweaks, button-fy the CTA link

mmmavis commented:
![button-ani](https://cloud.githubusercontent.com/assets/2896608/8446201/95bd29e0-1f5b-11e5-8460-03dd861d8e13.gif)
alrighty, I updated the PR again! Not quite sure how the CTA button should animate - I dropped the `box-shadow` and kept other interaction effects the same .

gvn commented:
Looks good! R+

mmmavis commented:
Coooooool, @gvn what's the release process?  Mind telling me what I should do next?

gvn commented:
I don't know actually. @thisandagain or @xmatthewx may be able to help you...

thisandagain commented:
Once this is merged (done) it will be automagically released to https://webmaker-desktop-staging.herokuapp.com

If everything checks out on staging, you just need to propose a PR to merge develop into master. Once that lands it will auto-deploy to https://beta.webmaker.org

xmatthewx commented:
@mmmavis - you can flag @vazquez when it's up on staging for a quick design review (automagic still pending). I'll be out Friday, back on Monday.

mmmavis commented:
@xmatthewx yea for sure, just curious how long does it usually take for things to show up on staging? Hopefully it will happen today so we can avoid pushing out a new release on Friday.
